src/twisted/trial/_synctest.py

```diff
@@ -1071,7 +1074,9 @@ def run(self, result):
 
         result.stopTest(self)
 
-    def addCleanup(self, f, *args, **kwargs):
+    def addCleanup(
+        self, f: Callable[_P, object], /, *args: _P.args, **kwargs: _P.kwargs
```

Comment:
this changes the defintion of addCleanup to match the TestCase.addCleanup - this is a breaking change where we should follow python or add `# type: ignore[override]`

Reply:
I prefer to follow Python stdlib API here, as commented above.

Reply:
I now have these methods marked as `type: ignore[override]`